### PR TITLE
[#160197650] Prometheus High Availability

### PR DESCRIPTION
**Pull request by mogds**
What
----

Prometheus currently runs in a single availability zone. This PR updates Prometheus so it's deployed to two availability zones and includes the following changes.

* `5b033803` Increases the instance count and specifies the availability zones.
* `8680a53e` Adds a custom Prometheus bosh release that enables [cross zone load balancing](https://github.com/alphagov/paas-prometheus-boshrelease/pull/1).
* `81829931` Replaces the nginx instance group with two zone specific instance groups. Updates the target group, listener rules and route 53 records so the correct nginx instance is registered.
* `9538aa85` Adds a Prometheus scale-down-dev.yml opsfile and configures the prometheus-deploy/upload-grafana-dashboards jobs to deploy based on slim_dev_deployment.
* `59c6fbba` Updates the relevant tests.

How to review
-------------

Code Review
Deploy with `SLIM_DEV_DEPLOYMENT` set to true/false

Who can review
--------------

Not @mogds @bandesz

### CODE REVIEW

File: manifests/cloud-config/operations.d/090-prometheus.yml [hunk 1]
```diff
@@ -1,9 +1,19 @@
...
+
+- type: replace
+  path: /vm_extensions/-
+  value:
+    name: prometheus_lb_z2
+    cloud_properties:
+      lb_target_groups:
+      - ((terraform_outputs_p8s_alertmanager_target_group_z2))
+      - ((terraform_outputs_p8s_grafana_target_group_z2))
+      - ((terraform_outputs_p8s_prometheus_target_group_z2))
```
Comment (bandesz) [inline manifests/cloud-config/operations.d/090-prometheus.yml]: Add an extra new line

File: manifests/prometheus/operations.d/000-custom-prometheus-bosh-release.yml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+---
+
+- type: replace
+  path: /releases/name=prometheus
+  value:
+    name: prometheus
+    version: 0.0.1548085118
+    url: https://s3-eu-west-1.amazonaws.com/gds-paas-build-releases/prometheus-0.0.1548085118.tgz
+    sha1: dead42725aeffcdaf9279818c3e8428e157feb90
```
Comment (bandesz) [inline manifests/prometheus/operations.d/000-custom-prometheus-bosh-release.yml]: new line

File: manifests/prometheus/operations.d/001-set-retention.yml [hunk 3]
```diff
@@ -1,5 +1,3 @@
----
```
Comment (bandesz) [inline manifests/prometheus/operations.d/001-set-retention.yml]: this should not be removed

File: manifests/prometheus/operations.d/400-set-instance-count.yml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+- type: replace
+  path: /instance_groups/name=firehose/instances
+  value: 2
+
+- type: replace
+  path: /instance_groups/name=firehose/azs
+  value:
+    - z1
+    - z2
+
```
Comment (bandesz) [inline manifests/prometheus/operations.d/400-set-instance-count.yml]: remove the extra spaces from the last line

File: manifests/cf-manifest/operations/scale-down-dev.yml [hunk 5]
```diff
@@ -32,3 +32,4 @@
 - type: replace
   path: /instance_groups/name=api/instances
   value: 1
+
```
Comment (bandesz) [inline manifests/cf-manifest/operations/scale-down-dev.yml]: this change should be removed

File: concourse/pipelines/create-cloudfoundry.yml [hunk 6]
```diff
@@ -1411,14 +1413,20 @@ jobs:
...
                 for f in "$(pwd)/paas-cf/manifests/prometheus/dashboards.d/"*.json; do
-                  sed "s/__SYSTEM_DNS_ZONE_NAME__/${SYSTEM_DNS_ZONE_NAME}/g" "${f}" \
-                  | sed "s/__DEPLOY_ENV__/${DEPLOY_ENV}/g" \
-                  | curl "https://grafana.${SYSTEM_DNS_ZONE_NAME}/api/dashboards/db" \
-                    --user "admin:$GRAFANA_PASS" \
-                    --header 'Content-Type: application/json' \
-                    --data "@-" \
-                    --fail \
-                    --include
+                  AZ_COUNT=2
```
Comment (bandesz) [inline concourse/pipelines/create-cloudfoundry.yml]: we should move these for lines outside of the outer for

File: manifests/prometheus/operations.d/000-custom-prometheus-bosh-release.yml [hunk 7]
```diff
@@ -0,0 +1,9 @@
+---
+
+- type: replace
+  path: /releases/name=prometheus
+  value:
+    name: prometheus
+    version: 0.0.1548085118
```
Comment (alext) [inline manifests/prometheus/operations.d/000-custom-prometheus-bosh-release.yml]: This looks to be a release build from a PR, not from master of our fork. Can we check whether that's the case, and update it to use a merged release version instead.

**Pull request merged**